controller/array_action/array_mediator_ds8k.py

```diff
@@ -167,20 +167,12 @@ def _connect(self):
                 raise array_errors.UnsupportedStorageVersionError(
                     self.version, self.SUPPORTED_FROM_VERSION
                 )
-        except (exceptions.ClientError, exceptions.Unauthorized) as e:
-            # BE7A002D=Authentication has failed because the user name and
-            # password combination that you have entered is not valid.
-            error_message = str(e.message).upper()
+        except exceptions.ClientException as ex:
+            error_message = str(ex.message).upper()
             if ERROR_CODE_INVALID_CREDENTIALS in error_message or KNOWN_ERROR_CODE_INVALID_CREDENTIALS in error_message:
                 raise array_errors.CredentialsError(self.service_address)
-            raise ConnectionError()
-        except exceptions.ClientException as e:
             logger.error(
-                'Failed to connect to DS8K array {}, reason is {}'.format(
-                    self.service_address,
-                    e.details
-                )
-            )
+                'Failed to connect to DS8K array {}, reason is {}'.format(self.service_address, ex.details))
             raise ConnectionError()
```

Comment:
while you're at it, could you change this to `raise ex`/`raise`, so we won't lose the exception info?

Reply:
We catch the `ClientException` exception and raise a different exception, `ConnectionError`.
If we decide to link them together (I'm totally into it), I suggest using `from`:
 ```suggestion
            raise ConnectionError() from ex
```

Reply:
not sure if `ConnectionInfo` adds any useful info

Reply:
due to the fact we have no handler for `ConnectionInfo` I changed it to `raise ex`